src/screen/HtmlScreen.js

```diff
@@ -206,6 +230,8 @@ class HtmlScreen extends RequestScreen {
 	 * @Override
 	 */
 	flip(surfaces) {
+		this.evaluateFavicon_();
```

Comment:
I'm not sure if I should call it that way or make it part of the promise of the flip.

```javascript
return this.evaluateFavicon_().then(
    super.flip(surfaces).then(() => {
			  utils.clearNodeAttributes(globals.document.documentElement);
			  utils.copyNodeAttributes(this.virtualDocument, globals.document.documentElement);
    })
);
```
or
```javascript
super.flip(surfaces).then(() => {
			utils.clearNodeAttributes(globals.document.documentElement);
			utils.copyNodeAttributes(this.virtualDocument, globals.document.documentElement);
      this.evaluateFavicon_();
});
```

Reply:
is `evaluateFavicon_` an async task?

Reply:
Yeah, returns a promise.

Reply:
Yeah, I saw it. But there's nothing async in the call chain for  `evaluateFavicon_`. So both of your options look valid. We probably don't need to make it return a promise since it doesn't do anything async.